src/detail/type_name.cpp

```diff
@@ -0,0 +1,71 @@
+/* Copyright 2017-2020 PaGMO development team
+
+This file is part of the PaGMO library.
+
+The PaGMO library is free software; you can redistribute it and/or modify
+it under the terms of either:
+
+  * the GNU Lesser General Public License as published by the Free
+    Software Foundation; either version 3 of the License, or (at your
+    option) any later version.
+
+or
+
+  * the GNU General Public License as published by the Free Software
+    Foundation; either version 3 of the License, or (at your option) any
+    later version.
+
+or both in parallel, as here.
+
+The PaGMO library is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received copies of the GNU General Public License and the
+GNU Lesser General Public License along with the PaGMO library.  If not,
+see https://www.gnu.org/licenses/. */
+
+#include <string>
+
+#if defined(__GNUC__) || (defined(__clang__) && !defined(_MSC_VER))
+
+// GCC demangle. This is available also for clang, both with libstdc++ and libc++.
+#include <cstdlib>
+#include <cxxabi.h>
+#include <memory>
+
+#endif
+
+#include <pagmo/detail/type_name.hpp>
+
+namespace pagmo
+{
+
+namespace detail
+{
+
+std::string demangle_from_typeid(const char *s)
+{
+#if defined(__GNUC__) || (defined(__clang__) && !defined(_MSC_VER))
```

Comment:
e che cazzo! ma mettersi daccordo no? in questo caso mi pare che MSVC faccia la cosa piu sana :)